var/spack/repos/builtin/packages/ffb/package.py

```diff
@@ -0,0 +1,163 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import os
+from spack import *
+
+
+class Ffb(Package):
+    """Computational Fluid Dynamics Software for aeroacoustics"""
+
+    homepage = "http://www.ciss.iis.u-tokyo.ac.jp/dl/index.php"
+    url      = "file://{0}/FrontFlow_blue.8.1.tar.gz".format(os.getcwd())
+    version('8.1', sha256='1ad008c909152b6c27668bafbad820da3e6ec3309c7e858ddb785f0a3d6e43ae')
+
+    patch('revocap_refiner.patch')
+
+    depends_on('mpi')
+    depends_on('blas')
+    depends_on('scalapack')
+
+    parallel = False
+
+    def install(self, spec, prefix):
+        d = find('.', 'make', recursive=True)
+        workdir = os.path.dirname(d[0])
+        popt = ['-P -traditional-cpp -Dcputime']
+        copt = ['-O3 ']
+        fopt = ['']
+        cflags = ['-O']
+        cxxflags = ['-O',  self.compiler.cxx_pic_flag]
+        fflags = ['']
+        ldshared = ['']
+        libs = ['-lstdc++']
+        incdir = spec['mpi'].headers.directories[0]
+        libdir = spec['mpi'].libs.directories[0]
+        if spec.satisfies('%gcc'):
+            fopt = ['-C ']
+            ldshared = ['g++ -shared -s']
+            fopt.append('-mcmodel=large')
+        if spec.satisfies('%intel'):
+            copt = ['']
+            fopt.append('-convert big_endian -mcmodel=large -shared-intel')
+            cflags = ['-O2']
+            cxxflags = ['-O2',  self.compiler.cxx_pic_flag]
+            fflags = ['-O2']
```

Comment:
This should all move to be defaults in the flag_handler method as well, with the exception of the fopt flags which appear to be necessary.